frontend/packages/console-shared/src/components/quick-starts/QuickStartsCatalogCard.tsx

```diff
@@ -0,0 +1,192 @@
+import { ArrowRightIcon } from '@patternfly/react-icons';
+import { RootState } from '@console/internal/redux';
+import {
+  QuickStart,
+  AllQuickStartStates,
+  QuickStartStatus,
+} from '@console/app/src/components/quick-starts/utils/quick-start-types';
+import {
+  getActiveQuickStartID,
+  getAllQuickStartStates,
+} from '@console/app/src/redux/reducers/quick-start-reducer';
+import * as QuickStartActions from '@console/app/src/redux/actions/quick-start-actions';
+import { getQuickStartStatus } from '@console/app/src/components/quick-starts/utils/quick-start-utils';
+import {
+  HIDE_QUICK_START_ADD_TILE_STORAGE_KEY,
+  HIDE_QUICK_START_DASHBOARD_TILE_STORAGE_KEY,
+} from './quick-starts-catalog-card-constants';
+
+import './QuickStartsCatalogCard.scss';
+
+type QuickStartsCatalogCardProps = {
+  quickStarts: QuickStart[];
+  storageKey: string;
+  onRemoveTile?: () => void;
+};
+
+type StateProps = {
+  activeQuickStartID?: string;
+  allQuickStartStates?: AllQuickStartStates;
+};
+
+type DispatchProps = {
+  setActiveQuickStart?: (quickStartID: string, totalTasks: number) => void;
+};
+
+type Props = QuickStartsCatalogCardProps & DispatchProps & StateProps;
+
+const QuickStartsCatalogCard: React.FC<Props> = ({
+  setActiveQuickStart,
+  quickStarts,
+  allQuickStartStates,
+  onRemoveTile,
+  storageKey,
+}) => {
+  const { t } = useTranslation();
+  const isQuickStartTileHidden = localStorage.getItem(storageKey) === 'true';
+  const [showTile, setShowTile] = React.useState<boolean>(!isQuickStartTileHidden);
+  const [isOpen, setOpen] = React.useState<boolean>(false);
+
+  const onRemove = () => {
+    localStorage.setItem(storageKey, 'true');
+    setShowTile(false);
+    onRemoveTile && onRemoveTile();
+  };
+
+  const onToggle = () => setOpen(!isOpen);
+
+  const actionDropdownItem = [
+    <DropdownItem
+      onClick={() => {
+        onRemove();
+      }}
+      key="action"
+      component="button"
+    >
+      {t('console-shared~Remove quick starts card from view')}
+      <div className="odc-quick-start-catalog-card__remove-help-text text-muted co-pre-wrap">
+        {t('console-shared~You will be able to access this information through the Help menu.')}
+      </div>
+    </DropdownItem>,
+  ];
+
+  const slicedQuickStarts: QuickStart[] = [];
+
+  quickStarts.forEach((quickStart) => {
+    if (
+      slicedQuickStarts.length < 3 &&
+      getQuickStartStatus(allQuickStartStates, quickStart.metadata.name) ===
+        QuickStartStatus.IN_PROGRESS
+    ) {
+      slicedQuickStarts.push(quickStart);
+    }
+  });
+
+  if (slicedQuickStarts.length !== 3) {
+    quickStarts.forEach((quickStart) => {
+      if (
+        slicedQuickStarts.length < 3 &&
+        getQuickStartStatus(allQuickStartStates, quickStart.metadata.name) ===
+          QuickStartStatus.NOT_STARTED
+      ) {
+        slicedQuickStarts.push(quickStart);
+      }
+    });
+  }
+
+  if (quickStarts.length > 0 && slicedQuickStarts.length === 0) {
```

Comment:
Why are you setting local storage to hide the card if there are no quick starts to show? That would mean the card would always be hidden even when a quick start is added later on.

Reply:
Better just render null when there's nothing to show.